src/fides/api/schemas/connection_configuration/connection_config.py

```diff
@@ -47,43 +49,34 @@ class Config:
         extra = Extra.forbid
 
 
-class TestStatus(Enum):
-    passed = "passed"
-    failed = "failed"
-    untested = "untested"
-
-    def str_to_bool(self) -> Optional[bool]:
-        """Translates query param string to optional/bool value
-        for filtering ConnectionConfig.last_test_succeeded field"""
-        if self == self.passed:
-            return True
-        if self == self.failed:
-            return False
-        return None
-
-
-class SystemType(Enum):
-    saas = "saas"
-    database = "database"
-    manual = "manual"
-    email = "email"
-
-
-class ConnectionSystemTypeMap(BaseModel):
+def mask_sensitive_fields(
+    connection_secrets: Dict[str, Any], secret_schema: Dict[str, Any]
+) -> Dict[str, Any]:
     """
-    Describes the returned schema for connection types
+    Mask sensitive fields in the given secrets based on the provided schema.
+    This function traverses the given secrets dictionary and uses the provided schema to
+    identify fields that have been marked as sensitive. The function replaces the sensitive
+    field values with a mask string ('********').
+    Note: This function modifies the input dictionary in-place.
+    Args:
+        connection_secrets (Dict[str, Any]): The secrets to be masked.
+        secret_schema (Dict[str, Any]): The schema defining which fields are sensitive.
+    Returns:
+        Dict[str, Any]: The secrets dictionary with sensitive fields masked.
     """
+    from pprint import pprint
+    pprint(connection_secrets)
 
-    identifier: Union[ConnectionType, str]
-    type: SystemType
-    human_readable: str
-    encoded_icon: Optional[str]
-
-    class Config:
-        """Use enum values and set orm mode"""
-
-        use_enum_values = True
-        orm_mode = True
+    print("secret_schema\n\n")
+    pprint(secret_schema)
+    if isinstance(connection_secrets, dict):
+        for key, val in connection_secrets.items():
+            prop = secret_schema["properties"].get(key, {})
+            if isinstance(val, dict):
+                mask_sensitive_fields(val, prop)
+            elif prop.get("sensitive", False):
+                connection_secrets[key] = "**********"
+    return connection_secrets
```

Comment:
Let's update this to omit fields that are not in the schema. This is a tricky one to catch but for OAuth connections, we store `access_token` and `refresh_token` in the `connectionconfig.secrets`. They're not in the schema because these aren't values we need the user to provide, they're added as part of the OAuth flow. Without this omission, these tokens are returned to the front-end in plain text.

Reply:
great catch. I'll update it to handle that now